packages/DropdownMenu/src/DropdownMenu.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Popover from "@paprika/popover";
+import ContentContainerStyled from "./ContentContainer.styles";
+import Confirmation from "./Confirmation/Confirmation";
+import Divider from "./Divider/Divider";
+import Trigger from "./Trigger/Trigger";
+import Item from "./Item/Item";
+
+const { alignTypes, oneOf, node, string, func } = PropTypes;
+
+const propTypes = {
+  /** Alignment of the dropdown menu */
+  align: oneOf(alignTypes),
+
+  /** Children should consist of <Dropdown.Item /> */
+  children: node.isRequired,
+
+  /** Optional custom classes */
+  className: string,
+
+  /** Render prop for rendering the trigger element that toggles the dropdown */
+  renderTrigger: func.isRequired,
+};
+
+const defaultProps = {
+  align: "bottom",
+  className: "",
+};
+
+const DropdownMenu = props => {
+  const { align, children, className } = props;
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isConfirming, setIsConfirming] = React.useState(false);
+  const [renderConfirmation, setRenderConfirmation] = React.useState(null);
+
+  const handleCloseMenu = () => {
+    setIsOpen(false);
+    setIsConfirming(false);
+    setRenderConfirmation(null);
+  };
+
+  const handleOpenMenu = () => {
+    setIsOpen(true);
+  };
+
+  const getTriggerStateAndHelpers = () => {
+    return {
+      isOpen,
+      handleOpenMenu,
+    };
+  };
+
+  const handleShowConfirmation = renderConfirmation => () => {
+    setIsConfirming(prevIsConfirmingState => !prevIsConfirmingState);
+    setRenderConfirmation(prevIsConfirmingState => (prevIsConfirmingState ? null : renderConfirmation));
+  };
+
+  const renderContent = () => {
+    if (isConfirming) {
+      return renderConfirmation(handleCloseMenu);
+    }
+
+    return (
+      <ContentContainerStyled isOpen={isOpen}>
+        {React.Children.toArray(children).map(child => {
+          if (child.type.componentType === "DropdownMenu.Item") {
+            if (child.props.renderConfirmation) {
+              return React.cloneElement(child, {
+                onShowConfirmation: handleShowConfirmation(child.props.renderConfirmation),
+              });
+            }
+            return React.cloneElement(child, {
+              onClose: handleCloseMenu,
+            });
+          }
+
+          return child;
+        })}
+      </ContentContainerStyled>
+    );
+  };
+
+  return (
+    <Popover
+      align={align}
+      className={className}
+      offset={4}
+      role={!isConfirming ? "menu" : null}
+      isOpen={isOpen}
+      onClose={() => {
+        if (!isConfirming) {
+          handleCloseMenu();
+        }
+      }}
+    >
+      <Popover.Trigger>{props.renderTrigger(getTriggerStateAndHelpers())}</Popover.Trigger>
+      <Popover.Content>{renderContent()}</Popover.Content>
```

Comment:
Since this part gets rendered into a portal, maybe we should have a mechanism to append a custom `className` (and maybe other testing attributes?) to this element? The app consumer will not be able to tag this otherwise which will be a problem for testing I think.  

I ran into this issue when using the `<Popover>` in Results.

Reply:
something like: `<Popover morePropsPopover={}` `morePropsTrigger={} />` 🙈

Reply:
hmm not sure i follow this. Do you mean adding additional trigger/content props to the popover component? Which are then spread on the `Trigger` and `Content` components respectively?

Reply:
The Trigger doesn't matter, it's contained in the root node for this component that already gets `moreProps` added to it, so that's sufficient.  But those `moreProps` will have no impact on the Popover.Content which is rendered in a portal `<div>`.  Not sure about how to best handle this though.... _could_ be more like this?
```jsx
<DropdownMenu>
  <DropdownMenu.Content className="something" data-qa-id="something" />
</DropdownMenu>
```

Reply:
😕 Hmm, Let's discuss this. I'm unsure if you mean we should use a permanent class name ie "popOver--content" or want to spread an additional props object here.

Reply:
I like your approach @mikrotron

Reply:
@mikrotron Ok perhaps can be added in next iteration as this PR is meant for port from acl-ui. If there is still alot more todo in this Pr i can address.

Reply:
Sure. Can you open an issue in GitHub to capture these kinds of "to-do" items.